test/Microsoft.ML.Benchmarks/Microsoft.ML.Benchmarks.csproj

```diff
@@ -4,6 +4,7 @@
     <OutputType>Exe</OutputType>
     <LangVersion>7.2</LangVersion>
     <StartupObject>Microsoft.ML.Benchmarks.Program</StartupObject>
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
```

Comment:
Let's not do this.  What is it required for? We typically don't do this in .NET Core projects (thus why it is not the default)

---
In reply to: [232782005](https://github.com/dotnet/machinelearning/pull/1599#discussion_r232782005) [](ancestors = 232782005)

Reply:
@eerhardt  When we do` dotnet run` the MSBuild copies to the output folder the dependencies that are somehow consumed in the code. If we don't use any type from BenchmarkDotNet.Diagnostics.Windows the .dll does not get copied and the console line argument parsing logic for `--profiler ETW` fails to load the .dll in a dynamic way ;(

Reply:
How does the dynamic loading of `--profiler ETW` work? This sounds like a bug in that logic. You should just be able to load an assembly, and it will load correctly if it is in the `.deps.json` file (which this one will be since you took a PackageReference on it).

Reply:
BTW - 

> When we do dotnet run the MSBuild copies to the output folder the dependencies that are somehow consumed in the code.

That is not true. Assemblies that are referenced through NuGet packages are not copied to the output folder, whether you use them in code or not. NuGet assemblies are loaded from the NuGet cache during `dotnet run`.

Reply:
@eerhardt I created a separate issue in BDN for that: https://github.com/dotnet/BenchmarkDotNet/issues/955